### PR TITLE
Make all Common_Spec tests pass for SQLSerever

### PR DESCRIPTION
**Pull request by AdRiley**
### Pull Request Description

Makes all tests in Common_Spec.enso pass for SQLServer. This doesn't cover full functionality just that set of tests.

Will refactor the database tests before proceeding further.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [ ] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [ ] Unit tests have been written where possible.

### CODE REVIEW

**Review by GregoryTravis — APPROVED** [review]

File: distribution/lib/Standard/Microsoft/0.0.0-dev/src/Internal/SQLServer_Dialect.enso [hunk 1]
```diff
@@ -17,6 +17,7 @@ import Standard.Database.DB_Column.DB_Column
 import Standard.Database.DB_Table.DB_Table
 import Standard.Database.Dialect
 import Standard.Database.Internal.Base_Generator
+import Standard.Database.Internal.Base_Generator.lift_binary_op
```
Comment (jdunkerley) [inline distribution/lib/Standard/Microsoft/0.0.0-dev/src/Internal/SQLServer_Dialect.enso]:
```suggestion
from Standard.Database.Internal.Base_Generator import lift_binary_op
```
Prefer this syntax for bring functions in but just a style thing.

File: distribution/lib/Standard/Microsoft/0.0.0-dev/src/Internal/SQLServer_Dialect.enso [hunk 2]
```diff
@@ -430,9 +461,7 @@ right = Base_Generator.lift_binary_op "RIGHT" str-> n->
 
 ## PRIVATE
 make_order_descriptor internal_column sort_direction text_ordering =
-    nulls = case sort_direction of
-        Sort_Direction.Ascending -> Nulls_Order.First
-        Sort_Direction.Descending -> Nulls_Order.Last
+    nulls = Nothing
```
Comment (jdunkerley) [inline distribution/lib/Standard/Microsoft/0.0.0-dev/src/Internal/SQLServer_Dialect.enso:464]: Are we going to solve this?

Standard way would be to have an extra order by to handle it.

File: test/Microsoft_Tests/src/SQLServer_Spec.enso [hunk 3]
```diff
@@ -78,13 +78,13 @@ create_connection_builder =
     connection_details.if_not_nothing <|
         _ -> Database.connect connection_details
 
-add_specs suite_builder =
+add_specs suite_builder  =
```
Comment (jdunkerley) [inline test/Microsoft_Tests/src/SQLServer_Spec.enso]:
```suggestion
add_specs suite_builder =
```

**Review by jdunkerley — APPROVED** [review]
A few little things but looks good

File: distribution/lib/Standard/Microsoft/0.0.0-dev/src/Internal/SQLServer_Dialect.enso (hunk 2 above)
Comment (AdRiley) [reply distribution/lib/Standard/Microsoft/0.0.0-dev/src/Internal/SQLServer_Dialect.enso:464]: Yes I was going to do it when I got to adding the test that tests for it.

File: test/Microsoft_Tests/src/SQLServer_Spec.enso (hunk 3 above)
Comment (AdRiley) [reply test/Microsoft_Tests/src/SQLServer_Spec.enso]: Fixed

File: distribution/lib/Standard/Microsoft/0.0.0-dev/src/Internal/SQLServer_Dialect.enso (hunk 1 above)
Comment (AdRiley) [reply distribution/lib/Standard/Microsoft/0.0.0-dev/src/Internal/SQLServer_Dialect.enso]: Done

**Pull request merged**